src/utils/loadFileHandler.js

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2015 - 2021, Nordic Semiconductor ASA
+ *
+ * All rights reserved.
+ *
+ * Use in source and binary forms, redistribution in binary form only, with
+ * or without modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions in binary form, except as embedded into a Nordic
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import fs from 'fs';
+import { deserialize } from 'bson';
+import { createInflateRaw } from 'zlib';
+import { Writable } from 'stream';
+
+const setupBuffer = async filename => {
+    const buffer = Buffer.alloc(370 * 1e6);
+    let size = 0;
+    const content = new Writable({
+        write(chunk, _encoding, callback) {
+            chunk.copy(buffer, size, 0);
+            size += chunk.length;
+            callback();
+        },
+    });
+
+    await new Promise((resolve, reject) => {
+        const readStream = fs.createReadStream(filename).on('error', err => {
+            reject(err);
+        });
+        const pipedStream = readStream.pipe(createInflateRaw());
+        pipedStream.on('error', err => {
+            reject(err);
+        });
+        pipedStream.pipe(content).on('finish', resolve);
+    });
```

Comment:
I read the discussion on Teams too late, so I am commenting here: 

What you are fighting with here is the fact that in the Node API each `pipe()` creates a new stream and `pipe` does not propagate errors from stream to stream. But in this case you want to handle an error that happens on any of the streams. 

There is actually a function to achieve what you want: [`pipeline()`](https://nodejs.org/docs/latest-v10.x/api/stream.html#stream_stream_pipeline_streams_callback). And there is [`promisify()`](https://nodejs.org/docs/latest-v10.x/api/util.html#util_util_promisify_original) to turn the callback variant into a Promise variant. So with the two combined you can condense that block to:
```suggestion
import { promisify } from 'util';
import { pipeline } from 'stream';

    await promisify(pipeline)(
        fs.createReadStream(filename),
        createInflateRaw(),
        content
    );
```
(Well, the imports have to be merged with the ones at the top of the file as you know)